### PR TITLE
[Experiment] Use `importlib` import mode for `pytest`

### PR DESCRIPTION
**Pull request by Taragolis**
Just an experiment whether or not airflow tests could use `importlib` mode instead of `prepend`

- [import mode](https://docs.pytest.org/en/latest/explanation/pythonpath.html#import-modes)
- [Choosing a test layout / import rules](https://docs.pytest.org/en/latest/explanation/goodpractices.html#choosing-an-import-mode)

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
interesting . I had no idea pytest is doing that.

**Comment by Taragolis** [comment]
> interesting . I had no idea pytest is doing that.

I've found myself it about week ago when try to find how to solve one issue.

**Comment by Taragolis** [comment]
🤔 🤔 🤔 

```console
        # Provide existent file path,
        # so loaded YAML file content should be used to call deserialize_model_dict(), rather than None.
        path = sys.path[0] + "/tests/providers/cncf/kubernetes/pod.yaml"
>       with open(path) as stream:
E       FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/bin/tests/providers/cncf/kubernetes/pod.yaml'
```

**Comment by potiuk** [comment]
> path = sys.path[0] + "/tests/providers/cncf/kubernetes/pod.yaml"

Yeah. Someone took advantage of assuming sys.path will be modified :)

File: pyproject.toml [hunk 1]
```diff
@@ -445,7 +445,15 @@ fixture-parentheses = false
 # * Disable `flaky` plugin for pytest. This plugin conflicts with `rerunfailures` because provide same marker.
 # * Disable `nose` builtin plugin for pytest. This feature deprecated in 7.2 and will be removed in pytest>=8
 # * And we focus on use native pytest capabilities rather than adopt another frameworks.
-addopts = "-rasl --verbosity=2 -p no:flaky -p no:nose --asyncio-mode=strict"
+addopts = [
+    "-rasl",
+    "--verbosity=2",
+    "-p", "no:flaky",
+    "-p", "no:nose",
+    "-p", "no:legacypath",
```
Comment (Taragolis) [inline pyproject.toml]: Well, I thought it doesn't add by a mistake, however seems like our tests use legacy LocalPath in some places

```console 
   @pytest.fixture(autouse=True)
    def refuse_to_run_test_from_wrongly_named_files(request):
>       dirname: str = request.node.fspath.dirname
E       AttributeError: 'Function' object has no attribute 'fspath'
```

**Pull request closed without merging**